packages/rln/src/contract/rln_base_contract.ts

```diff
@@ -84,6 +89,8 @@ export class RLNBaseContract {
     instance.minRateLimit = ethers.BigNumber.from(min).toNumber();
     instance.maxRateLimit = ethers.BigNumber.from(max).toNumber();
 
+    RLNBaseContract.Q = (await instance.contract.Q()) || DEFAULT_Q;
```

Comment:
I advice to implement here a method either `get Q` or `init Q`
which will do `try { await instance.contract.Q() } catch` and retreat to `defaul`